services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaService.kt

```diff
@@ -0,0 +1,52 @@
+package fi.hel.haitaton.hanke.permissions
+
+import fi.hel.haitaton.hanke.application.ApplicationData
+import fi.hel.haitaton.hanke.application.Contact
+import java.util.UUID
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class HankeKayttajaService(
+    private val hankeKayttajaRepository: HankeKayttajaRepository,
+    private val kayttajaTunnisteRepository: KayttajaTunnisteRepository,
+) {
+
+    @Transactional
+    fun saveNewTokensFromApplication(applicationData: ApplicationData, hankeId: Int) {
+        // Contacts new for this application
+        val newContacts =
+            applicationData
+                .customersWithContacts()
+                .flatMap { it.contacts }
+                .filter { !it.email.isNullOrBlank() && !it.name.isNullOrBlank() }
+
+        // Emails that already exist for the hanke, from either the hanke contacts or from other
```

Comment:
I think these comments are redundant as the code is self explanatory.

Reply:
Comment removed.